home-manager/install.nix

```diff
@@ -13,7 +13,9 @@ runCommand "home-manager-install" {
       echo "Creating initial Home Manager configuration..."
 
       mkdir -p "$(dirname "$confFile")"
-      cat > $confFile <<EOF
+      
+      if [[ -f /etc/NIXOS ]] || grep --regexp "^ID=nixos$" /etc/os-release > /dev/null 2>&1; then
```

Comment:
I think that this check is too brittle. What if, for whatever reason, NixOS no longer created the `/etc/NIXOS` or `/etc/os-release` files? A pretty big what-if, but I really don't like the idea of relying on files to determine this kind of thing.

IMHO, there's no need to be smart and try to detect what the user wants. I think a better solution is to add this setting to the default configuration. Something along the lines of:

```
<...>

# The host operating system that Home Manager will be installed on. This enables
# per-OS workarounds and tweaks to increase the general cohesiveness of the
# system. See `man home-configuration.nix` for possible values.
home.targetOperatingSystem = "nixos";

<...>
```

Reply:
I like this approach, what do you think @rycee ?